### ISSUE TITLE
grpo+cosine 配置 solution参数无法传入问题？

### ISSUE THREAD

**Issue by NiceKatan** [modelscope/ms-swift#7851]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

CUDA_VISIBLE_DEVICES=1,2,3 NPROC_PER_NODE=3 swift rlhf \
  --rlhf_type grpo \
  --model_type qwen3\
  --template qwen3_nothinking\
  --model Qwen3-4B-Instruct-sft \
  --dataset rl_train_data.jsonl \
  --val_dataset rl_train_data_val.jsonl \
  --external_plugins examples/train/grpo/plugin/plugin.py \
  --reward_funcs cosine \
  --loss_type grpo \
  --torch_dtype float16\
  --temperature 0.2 \
  --top_p 0.95 \
  --max_completion_length 128 \
  --per_device_train_batch_size 1 \
  --num_generations 3 \
  --gradient_accumulation_steps 8 \
  --max_length 2048 \
  --learning_rate 1e-5 \
  --num_train_epochs 1 \
  --logging_steps 10 \
  --eval_steps 100 \
  --save_steps 200 \
  --save_total_limit 3 \
  --report_to tensorboard \
  --output_dir qwen3_sft_grpo_acc

上面是运行命令

{"messages": [{"role": "system", "content": "请你扮演一位金融和会计领域专家，你会面临用户提出的一些问题，你要给出解决问题的思考过程和最终答案。你要首先在头脑中思考推理过程，然后向用户提供答案。最后，答案要用 $\\boxed{}$的形式输出。"}, {"role": "user", "content": "你是一个金融专家，擅长使用极其简练的语言进行逻辑推理。\n请根据提供的金融知识，给出精简的思维链推理并选择正确答案。\n关于期货市场交易的理解，哪一项是正确的（）？\nA. 同时进行的有协议成交和标的交割\nB. 无论交易对象价格如何波动，都无法影响交易者的利润或损失\nC. 买家和卖家的交易依据仅是他们对市场未来的预测评估\nD. 交易双方承受的利率和汇率风险微小"}], "solution": "C"}

上面是数据集格式，数据集加载过程后可以显示message 与 solution 字段

[INFO:swift] train_dataset: Dataset({
    features: ['messages', 'solution'],
    num_rows: 5703
})
[INFO:swift] val_dataset: Dataset({
    features: ['messages', 'solution'],
    num_rows: 99
})

请问为什么 reward 中 solution 参数一直为None，计算acc时solution无法透传

**Comment by hjh0119** [comment]
plz provide env infos and minimal reproduce script

**Comment by NiceKatan** [comment]
> plz provide env infos and minimal reproduce script请提供环境信息和最小复现脚本

jsonl数据样例：

[rl_train_data_test.txt](https://github.com/user-attachments/files/24788757/rl_train_data_test.txt)

复现脚本：

CUDA_VISIBLE_DEVICES=1,2,3 NPROC_PER_NODE=3 swift rlhf
--rlhf_type grpo
--model_type qwen3
--template qwen3_nothinking
--model Qwen3-4B-Instruct-sft // 本地部署
--dataset rl_train_data.jsonl
--external_plugins examples/train/grpo/plugin/plugin.py
--reward_funcs cosine
--loss_type grpo
--torch_dtype float16
--temperature 0.2
--top_p 0.95
--max_completion_length 128
--per_device_train_batch_size 1
--num_generations 3
--gradient_accumulation_steps 8
--max_length 2048
--learning_rate 1e-5
--num_train_epochs 1
--logging_steps 10
--eval_steps 100
--save_steps 200
--save_total_limit 3
--report_to tensorboard
--output_dir qwen3_sft_grpo_acc

配置环境：

[requirements.txt](https://github.com/user-attachments/files/24788917/requirements.txt)

**Comment by hjh0119** [comment]
更新下源码再试试

**Comment by NiceKatan** [comment]
> 更新下源码再试试

已解决，非常感谢
